### PR TITLE
Surrogate Branch

### PR DESCRIPTION
**Pull request by lzn87**
- Refactored ```buildings900K.py```. It now supports data loading for surrogate model training, which expects the index file that has one building per row and ignores seq_ptr. It randomly selects a sequence of length = pred_len on the fly, where the starting index is in range [1, 8736 - pred_len] (8736 is the total number of hours in a year and we skip the first hour of the first day due to missing weather data). More details on creating simcap index files can be found ```script/create_index_files_simcap.py```.
- Added a ```BaseSurrogateModel``` abstract class and implemented surrogate model class ```ResNet``` that makes prediction on single timestamp with residual connections. Added configuration files for small resnets ```ResNet-S``` and ```ResNet-Text-S``` with text embedding. 
- Added scripts for caption generation, post processing, BERT embedding generation in parallel.
- Fixed small bug in validation function.
- Added script for surrogate model evaluation
- Added weight initialization function in surrogate model abstract class, and xavier_uniform initialization for linear ResNet
- Removed random seed in Builgings900k surrogate mode
- Removed lat/lon and building type embedding for surrogate models
- Added scaler transform for resnet surrogate model
- Fixed inverse transform bug

### CODE REVIEW

File: buildings_bench/models/resnet.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                pred_len=1,
+                continuous_head='mse',
+                continuous_loads=True):
+        
+        super(ResNet,self).__init__(pred_len, continuous_loads)
+        self.continuous_head = continuous_head
+        out_dim = 1 if self.continuous_head == 'mse' else 2
+        self.logits = nn.Linear(hidden_dims[-1], out_dim)
+        
+        self.building_embedding = nn.Embedding(2, 32)
```
Comment (pemami4911) [inline buildings_bench/models/resnet.py]: I think moving forward let's remove these features-- "building_embedding" (which is binary 0 or 1 for res/com), "lat_embedding" , and "lon_embedding". These infos are going to be in the building characteristic feature vector /text caption now, unlike in BuildingsBench where we don't have this

File: scripts/create_index_files_simcap.py [hunk 2]
```diff
@@ -36,39 +36,31 @@ def main(args):
...
-        train_idx = set(bd_ids[mask])
-        test_idx = set(bd_ids[~mask])
-        
-        if 'amy2018' in by:
-            train_hours = int((train_amy2018_timerange[1] - train_amy2018_timerange[0]).total_seconds() / 3600)
-            print(f'AMY2018 train hours: {train_hours}')
 
-            val_hours = int((val_timerange[1] - val_timerange[0]).total_seconds() / 3600)
-            print(f'AMY2018 Val hours: {val_hours}')
+        # randomly split train (60%) and val (20%) test (20%) sets
```
Comment (pemami4911) [inline scripts/create_index_files_simcap.py:56]: this is OK for now, for "i.i.d. generalization" experiments. We need to also create some "OOD generalization" test splits for CLP^2, where we identify attributes we want to use to test systematic generalization (withholding combinations). My suggestion is to pick ranges of values (continuous) or certain buckets (discrete) from two attributes and withhold the intersection. I can help with this.

File: scripts/surrogate_eval.py [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+        transform = lambda x: x
+        inverse_transform = load_transform.undo_transform
+    else: # Continuous unscaled values
+        transform = lambda x: x
+        inverse_transform = lambda x: x
+
+    model.eval()
+    step = 0
+
+    bd_types = ['FullServiceRestaurant', 'Hospital', 'LargeHotel', 'LargeOffice',
```
Comment (pemami4911) [inline scripts/surrogate_eval.py:166]: This is also OK for now, but this is a good reminder that we need to make progress on writing code for the various other qualitative evaluation metrics we will use

File: buildings_bench/models/resnet.py (hunk 1 above)
Comment (lzn87) [reply buildings_bench/models/resnet.py]: Removed!

**Comment by pemami4911** [comment]
Thanks!
